### PR TITLE
Implement Roles User Stories

### PR DESCRIPTION
**Pull request by markodragicevicct**
Signed-off-by: Marko Dragicevic <marko.dragicevic@comtrade.com>

### CODE REVIEW

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/RolePermissionToolbar.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package org.eclipse.kapua.app.console.client.role;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/RolePermissionToolbar.java]: Please add copyright headers.

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleSubjectGrid.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
+package org.eclipse.kapua.app.console.client.role;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleSubjectGrid.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabSubjectGrid.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+    public RoleTabSubjectGrid(GwtSession gwtSession) {
+        super("Subject", new KapuaIcon(IconSet.SUPPORT));
+        rolePermissionGrid = new RoleSubjectGrid(null, gwtSession);
+
+    }
+
+    @Override
+    protected void doRefresh() {
+        if (selectedEntity != null) {
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabSubjectGrid.java]: I think that is something that was thought differently :)

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/dialog/RolePermissionAddDialog.java [hunk 4]
```diff
@@ -0,0 +1,102 @@
+package org.eclipse.kapua.app.console.client.role.dialog;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/dialog/RolePermissionAddDialog.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/dialog/RolePermissionDeleteDialog.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
+package org.eclipse.kapua.app.console.client.role.dialog;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/dialog/RolePermissionDeleteDialog.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/server/GwtAccessInfoServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,73 @@
+package org.eclipse.kapua.app.console.server;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/server/GwtAccessInfoServiceImpl.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/shared/model/GwtAccessRole.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
+package org.eclipse.kapua.app.console.shared.model;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/shared/model/GwtAccessRole.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/shared/model/authorization/GwtAccessRoleQuery.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package org.eclipse.kapua.app.console.shared.model.authorization;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/shared/model/authorization/GwtAccessRoleQuery.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/shared/model/authorization/GwtRolePermissionCreator.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
+package org.eclipse.kapua.app.console.shared.model.authorization;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/shared/model/authorization/GwtRolePermissionCreator.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/shared/service/GwtAccessInfoService.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
+package org.eclipse.kapua.app.console.shared.service;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/shared/service/GwtAccessInfoService.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabSubjectGrid.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
+package org.eclipse.kapua.app.console.client.role;
```
Comment (Coduz) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabSubjectGrid.java]: Please add copyrights headers.

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabPermissionGrid.java [hunk 12]
```diff
@@ -15,16 +15,21 @@
...
 import com.google.gwt.user.client.Element;
 
 public class RoleTabPermissionGrid extends KapuaTabItem<GwtRole> {
 
-    RolePermissionGrid rolePermissionGrid = new RolePermissionGrid(null, null);
+    RolePermissionGrid rolePermissionGrid;
 
-    public RoleTabPermissionGrid() {
+    public RoleTabPermissionGrid(EntityView<GwtRolePermission> entityView, GwtSession session) {
         super("Permission", new KapuaIcon(IconSet.TASKS));
```
Comment (lorthirk) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabPermissionGrid.java]: Could you add a localizable string in GwtRoleMessages?

File: console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabSubjectGrid.java [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+import org.eclipse.kapua.app.console.shared.model.GwtSession;
+import org.eclipse.kapua.app.console.shared.model.authorization.GwtRole;
+import com.google.gwt.user.client.Element;
+
+public class RoleTabSubjectGrid extends KapuaTabItem<GwtRole> {
+
+    RoleSubjectGrid rolePermissionGrid;
+
+    public RoleTabSubjectGrid(GwtSession gwtSession) {
+        super("Subject", new KapuaIcon(IconSet.SUPPORT));
```
Comment (lorthirk) [inline console/src/main/java/org/eclipse/kapua/app/console/client/role/RoleTabSubjectGrid.java]: Could you add a localizable string in GwtRoleMessages?

**Pull request merged**